rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java

```diff
@@ -137,40 +148,74 @@ private Optional<Short> getAvailableBucketWithLessUsedGas(long txGasLimit) {
 
 
     private TransactionBucket getBucketCandidates(Transaction tx, Set<ByteArrayWrapper> newReadKeys, Set<ByteArrayWrapper> newWrittenKeys) {
-        Set<Short> bucketCandidates = new HashSet<>();
-
-        getBucketBySender(tx).ifPresent(bucketCandidates::addAll);
+        Optional<Short> bucketCandidates = getBucketBySender(tx);
 
-        if (bucketCandidates.size() > 1) {
+        if (bucketCandidates.isPresent() && isSequentialId(bucketCandidates.get())) {
             return getSequentialBucket();
         }
 
-        // read - written
-        getCandidatesByReadWriteKeyCollisions(newReadKeys, bucketCandidates);
 
-        if (bucketCandidates.size() > 1) {
-            return getSequentialBucket();
+        // read - written
+        Optional<Short> bucketCandidate = bucketCandidates;
```

Comment:
Is this really needed?

Reply:
Nope, my bad. It stayed there after the refactor.